### PR TITLE
TSLint - strict boolean expressions - 4

### PR DESCRIPTION
**Pull request by digeff**
Prepare to enable the strict boolean expressions from TSLint. Stop using non-boolean values as booleans.
(This change will be split among several PRs)

We are replacing the expressions flagged by the rule with: https://github.com/Microsoft/vscode-chrome-debug-core/blob/v2/src/chrome/utils/typedOperators.ts

### CODE REVIEW

File: src/chrome/consoleHelper.ts [hunk 1]
```diff
@@ -110,20 +113,20 @@ function resolveParams(args: CDTP.Runtime.RemoteObject[], skipFormatSpecifiers?:
...
-                currentCollapsedStringArg += (currentCollapsedStringArg ? ' ' + formatted : formatted);
+                currentCollapsedStringArg += (currentCollapsedStringArg === '' ? ' ' + formatted : formatted);
             }
-        } else if (formatSpec) {
+        } else if (isNotEmpty(formatSpec)) {
             // `formatted` is an object - split currentCollapsedStringArg around the current formatSpec and add the object
             const curSpecIdx = currentCollapsedStringArg.indexOf('%' + formatSpec);
             const processedPart = currentCollapsedStringArg.slice(0, curSpecIdx);
-            if (processedPart) {
+            if (processedPart !== '') {
```
Comment (mrcrane) [inline src/chrome/consoleHelper.ts]: Use isEmpty?

File: src/utils.ts [hunk 2]
```diff
@@ -193,31 +196,27 @@ export function forceForwardSlashes(aUrl: string): string {
...
-
-    if (aPath.match(/file:\/\/\/[A-Za-z]:/)) {
+    if (hasMatches(aPath.match(/file:\/\/\/[A-Za-z]:/))) {
         const prefixLen = 'file:///'.length;
         aPath =
             'file:///' +
             aPath[prefixLen].toLowerCase() +
             aPath.substr(prefixLen + 1);
-    } else if (aPath.match(/^[A-Za-z]:/)) {
+    } else if (aPath.match(/^[A-Za-z]:/) !== null) {
```
Comment (mrcrane) [inline src/utils.ts]: Use isNull?

File: src/utils.ts [hunk 3]
```diff
@@ -240,11 +239,11 @@ export function stripTrailingSlash(aPath: string): string {
  * when passing on a failure from a Promise error handler.
  * @param msg - Should be either a string or an Error
  */
-export function errP(msg: string | Error): Promise<never> {
+export function errP(msg?: string | Error): Promise<never> {
     const isErrorLike = (thing: any): thing is Error => !!thing.message;
 
     let e: Error;
-    if (!msg) {
+    if (msg === undefined) {
```
Comment (mrcrane) [inline src/utils.ts]: use isUndefined?

File: src/utils.ts [hunk 4]
```diff
@@ -489,7 +457,7 @@ export function pathToRegex(aPath: string, guid = ''): string {
         aPath = escapeRegexSpecialChars(fileUrlPrefix) + aPath;
     }
 
-    if (guid) {
+    if (guid !== '') {
```
Comment (mrcrane) [inline src/utils.ts]: use isEmpty?

File: src/chrome/consoleHelper.ts (hunk 1 above)
Comment (digeff) [reply src/chrome/consoleHelper.ts]: Changed to isNotEmpty

File: src/utils.ts (hunk 2 above)
Comment (digeff) [reply src/utils.ts]: Changed to hasMatches

File: src/utils.ts (hunk 4 above)
Comment (digeff) [reply src/utils.ts]: Changed to isNotEmpty

File: src/utils.ts (hunk 3 above)
Comment (digeff) [reply src/utils.ts]: Changed method to only accept strings, and changed to isEmpty

**Pull request merged**